packages/matchbox/src/components/ListBox/ListBox.tsx

```diff
@@ -28,40 +34,77 @@ const StyledWrapper = styled(Box)`
   ${system}
 `;
 
-const StyledChevron = styled(KeyboardArrowDown)`
+const StyledChevron = styled(KeyboardArrowDown)<{ $disabled?: boolean }>`
   ${chevron}
 `;
 
-const StyledButton = styled(Button)`
+type StyledButtonProps = React.ComponentPropsWithRef<'button'> &
+  ButtonProps & {
+    $hasError?: boolean;
+  };
+
+const StyledButton = styled(Button)<StyledButtonProps>`
   text-align: left;
-  font-size: ${props => props.theme.fontSizes['200']};
-  font-weight: ${props => props.theme.fontWeights['medium']};
-  background: ${props => (props.disabled ? props.theme.colors.gray['200'] : '')};
-  padding-right: ${props => props.theme.space[650]};
-  border: ${props =>
-    props.hasError ? `1px solid ${props.theme.colors.red[700]}` : `${props.theme.borders[400]}`};
+  font-size: ${(props) => props.theme.fontSizes['200']};
+  font-weight: ${(props) => props.theme.fontWeights['medium']};
+  background: ${(props) => (props.disabled ? props.theme.colors.gray['200'] : '')};
+  padding-right: ${(props) => props.theme.space[650]};
+  border: ${(props) =>
+    props.$hasError ? `1px solid ${props.theme.colors.red[700]}` : `${props.theme.borders[400]}`};
   &:hover {
-    background: ${props => (props.disabled ? props.theme.colors.gray['200'] : 'transparent')};
+    background: ${(props) => (props.disabled ? props.theme.colors.gray['200'] : 'transparent')};
   }
   ${focusOutline()}
 `;
 
-const ListBoxButton = React.forwardRef(function ListBoxButton(props, userRef) {
+const ListBoxButton = React.forwardRef<HTMLButtonElement, StyledButtonProps>(function ListBoxButton(
+  props,
+  userRef,
+) {
   return (
     <StyledButton {...props} ref={userRef}>
       {props.children}
     </StyledButton>
   );
-});
+}) as React.ForwardRefExoticComponent<StyledButtonProps>;
```

Comment:
Don't think you need this line unless you're extending `ForwardRefExoticComponent`